Source/player.cpp

```diff
@@ -1486,7 +1508,13 @@ void StartPlrBlock(int pnum, direction dir)
 	if (!(plr[pnum]._pGFXLoad & PFILE_BLOCK)) {
 		LoadPlrGFX(pnum, PFILE_BLOCK);
 	}
-	NewPlrAnim(pnum, plr[pnum]._pBAnim[dir], plr[pnum]._pBFrames, 2, plr[pnum]._pBWidth);
+
+	int skippedAnimationFrames = 0; // Block can start with Frame 1 if Player 2 hits Player 1. In this case Player 1 will not call again ProcessPlayerAnimation.
+	if (plr[pnum]._pIFlags & ISPL_FASTBLOCK) {
+		skippedAnimationFrames = (plr[pnum]._pBFrames - 1); // ISPL_FASTBLOCK means there is only one AnimationFrame.

```

Comment:
Comment says 1 total, but code says one less. Is this because block just happens to be 2 frames normally?

Reply:
_pBFrames is the total number of frames. not the current.
So we say all frames are skipped expect frame 1 😉.
With the warrior and normal shield the total number of frames are only 2 frames 😅

But I looked again at the relevant logic in `PM_DoBlock`

`	if (plr[pnum]._pIFlags & ISPL_FASTBLOCK && plr[pnum]._pAnimFrame != 1) {
		plr[pnum]._pAnimFrame = plr[pnum]._pBFrames;
	}`

But perhaps I'm too aggressive here. The logic kicks only in if frame != 1. So if Player1 hits Player2, it will skip all Frames (cause Player2 runs ProcessPlayerAnimations in same gametick). But if Player 2 hits Player 1 it wouldn't.
So maybe -2? 🤔

Reply:
Dont forget that blocking utilizes the animation delay system: the first frame is actually shown during 3 in-game frames, so the total block duration normally is 4 frames.